### ISSUE TITLE
register function does not save profile data in Android

### ISSUE THREAD

**Issue by fdobre** [colorfy-software/react-native-gigya-sdk#4]
Steps to reproduce:

use GigyaSdk.registerAccount<void | any>(params.email, params.password, params); and provide additional params as
{profile: {firstName: 'Test', lastName: 'TestLastName'}}

For iOS everything works and all the data is saved.

For Android the BE response is:

```
 { type: 'invalidParameter',
  payload: 
   { callId: 'fc8568e5783b45eb93b6e80fdaeb0e3f',
     errorCode: 400006,
     errorDetails: 'Invalid argument: Invalid JSON in \'profile\'',
     errorMessage: 'Invalid parameter value',
     apiVersion: 2,
     statusCode: 400,
     statusReason: 'Bad Request',
     time: '2022-06-17T15:01:39.888Z' },
  message: '{\n  "callId": "fc8568e5783b45eb93b6e80fdaeb0e3f",\n  "errorCode": 400006,\n  "errorDetails": "Invalid argument: Invalid JSON in \'profile\'",\n  "errorMessage": "Invalid parameter value",\n  "apiVersion": 2,\n  "statusCode": 400,\n  "statusReason": "Bad Request",\n  "time": "2022-06-17T15:01:39.888Z"\n}\n',
```

Issue seems to be related to the JSON formatting in the library files:

```
@ReactMethod
    public void registerAccount(final String email, final String password, final String params, final Promise promise) {
        try {
            mGigya.register(email, password, parseParamsString(params), new GigyaLoginHandler(promise));
        } catch(Exception e) {
            promise.reject("registerAccount", String.valueOf(e));
        }

    }

 ...
  public static HashMap<String, Object> parseParamsString(final String params) throws Exception{
        return new Gson().fromJson(params, new TypeToken<HashMap<String, Object>>() {}.getType());
    }

 ....
import type { GigyaSdkRegisteredAccountType } from '../types';
export default function <ParamsType extends Record<string, unknown>>(email: string, password: string, params?: ParamsType): Promise<GigyaSdkRegisteredAccountType>;
```

**Comment by CharlesMangwa** [comment]
Hey @fdobre! I'd have to investigate as we haven't faced this ourselves. In the meantime, as mentioned in #3, you can consider using the [REST API equivalent](https://help.sap.com/docs/SAP_CUSTOMER_DATA_CLOUD/8b8d6fffe113457094a17701f63e3d6a/1fe26c820cd145cd8c927a497c33d935.html?locale=en-US):

```tsx
await GigyaSdk.sendApiCall('accounts.register', parameters)
```

**Comment by CharlesMangwa** [comment]
Hi @fdobre! Just to let you know that we've just released [`v1.2.0`](https://github.com/colorfy-software/react-native-gigya-sdk/releases/tag/v1.2.0) with a fix for this issue. We were able to successfully call the following method on both iOS and Android in the [example app](https://github.com/colorfy-software/react-native-gigya-sdk/blob/main/example/src/App.tsx#L48-L50):

```ts
await GigyaSdk.registerAccount(CONFIG.EMAIL, CONFIG.PASSWORD, {
  profile: { firstName: 'Test', lastName: 'User' },
})
```

Let us know if you still have any issues with it!
